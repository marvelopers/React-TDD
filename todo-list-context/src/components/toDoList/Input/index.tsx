import React from "react";
import styled from "styled-components";

interface InputProps {
  value: string;
  placeholder?: string;
  onChange: (toDo: string) => void;
}

const Input = ({ value, placeholder, onChange }: InputProps) => (
  <InputBox
    value={value}
    placeholder={placeholder}
    onChange={(e) => onChange(e.target.value)}
  />
);

export default Input;

const InputBox = styled.input`
  width: 100%;
  padding: 10px 10px;
  border-radius: 8px;
  border: 1px solid #bdbdbd;
  outline: none;
`;
