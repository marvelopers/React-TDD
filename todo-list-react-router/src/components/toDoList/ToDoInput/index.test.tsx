import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import 'jest-styled-components';

import Input from './index';

describe('<ToDoInput/>', () => {
  it('renders component correctly', () => {
    const { container } = render(<Input value="default value" onChange={() => jest.fn()} />);

    const input = screen.getByDisplayValue('default value');
    expect(input).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('renders placeholder correctly', () => {
    render(<Input value="default value" onChange={() => jest.fn()} />);

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요 :)');
    expect(input).toBeInTheDocument();
  });

  it('change the data', () => {
    render(<Input value="study tdd" onChange={() => jest.fn()} />);

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요 :)') as HTMLInputElement;
    fireEvent.change(input, { target: { value: 'study tdd' } });
    expect(input.value).toBe('study tdd');
  });
});
