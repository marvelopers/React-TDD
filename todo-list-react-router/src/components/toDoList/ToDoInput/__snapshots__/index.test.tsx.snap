// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDoInput/> renders component correctly 1`] = `
.c0 {
  width: 100%;
  padding: 10px 10px;
  border-radius: 8px;
  border: 1px solid #bdbdbd;
  outline: none;
}

<div>
  <input
    class="c0"
    placeholder="할 일을 입력해 주세요 :)"
    value="default value"
  />
</div>
`;
